cmd/zc_traverser_local.go

```diff
@@ -63,18 +155,53 @@ func (t *localTraverser) traverse(processor objectProcessor, filters []objectFil
 
 	// if the path is a single file, then pass it through the filters and send to processor
 	if isSingleFile {
-		t.incrementEnumerationCounter()
-		err = processIfPassedFilters(filters, newStoredObject(singleFileInfo.Name(),
+		if t.incrementEnumerationCounter != nil {
+			t.incrementEnumerationCounter()
+		}
+
+		return processIfPassedFilters(filters, newStoredObject(singleFileInfo.Name(),
 			"", // relative path makes no sense when the full path already points to the file
 			singleFileInfo.ModTime(), singleFileInfo.Size(), nil, blobTypeNA), processor)
-		return
-
 	} else {
 		if t.recursive {
-			// We want to re-queue symlinks up in their evaluated form because filepath.Walk doesn't evaluate them for us.
+			processFile := func(filePath string, fileInfo os.FileInfo, fileError error) error {
+				if fileError != nil {
+					glcm.Info(fmt.Sprintf("Accessing %s failed with error: %s", filePath, fileError))
+					return nil
+				}
+
+				if fileInfo.IsDir() {
+					return nil
+				}
+
+				relPath := strings.TrimPrefix(strings.TrimPrefix(cleanLocalPath(filePath), cleanLocalPath(t.fullPath)), "/")
+				if !t.followSymlinks && fileInfo.Mode()&os.ModeSymlink != 0 {
+					glcm.Info(fmt.Sprintf("Skipping over symlink at %s because --follow-symlinks is false", filepath.Join(t.fullPath, relPath)))
+					return nil
+				}
+
+				return processIfPassedFilters(filters,
+					newStoredObject(
+						fileInfo.Name(),
+						relPath,
+						fileInfo.ModTime(),
+						fileInfo.Size(),
+						nil,
+						blobTypeNA),
+					processor)
+			}
+
+			if t.followSymlinks {
+				return WalkWithSymlinks(t.fullPath, processFile)
+			} else {
+				return filepath.Walk(t.fullPath, processFile)
+			}
+
+			/*// We want to re-queue symlinks up in their evaluated form because filepath.Walk doesn't evaluate them for us.
```

Comment:
Please remove this commented out block. We do not leave big commented out blocks like this in the code. (We just use file history in Git if we need the old version)